packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -36,14 +36,19 @@ const ADD_LISTENER_METHODS = [
   'prependOnceListener' as 'prependOnceListener',
 ];
 
-export abstract class BaseContextManager implements ContextManager {
+export abstract class AbstractAsyncHooksContextManager
+  implements ContextManager {
   abstract active(): Context;
 
   abstract with<T extends (...args: unknown[]) => ReturnType<T>>(
```

Comment:
Think you can do this if you want to avoid duplicating the complex type, but I'm not actually 100% sure... Either way, this isn't a blocker

```suggestion
  abstract with: ContextManager["with"];
```

Reply:
This doesn't seem to work as now typescript complaints

> Class 'AbstractAsyncHooksContextManager' defines instance member property 'with', but extended class 'AsyncHooksContextManager' defines it as instance member function.

Reply:
oh well